src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -10,14 +10,15 @@
 
 const MIGRATION_STATUS = {
   NOT_MIGRATED: 'NOT_MIGRATED',
-  MIGRATED: 'MIGRATED'
+  MIGRATED: 'MIGRATED',
+  MIGRATING: 'MIGRATING'
 };
 
 /**
  *
  * @returns {UseLocalDatabaseHook}
  */
-const useLocalDatabaseHook = () => {
+const useLocalDatabaseHook = (withMigration = false) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The modification of `useLocalDatabaseHook` to accept an optional parameter `withMigration` is a good approach. This allows the function to be more flexible and can be used in scenarios where database migration is not required.

```diff
- const useLocalDatabaseHook = () => {
+ const useLocalDatabaseHook = (withMigration = false) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->